pkg/controllers/clusterresourceplacement/placement_controller.go

```diff
@@ -333,3 +339,153 @@ func (r *Reconciler) updatePlacementAppliedCondition(placement *fleetv1alpha1.Cl
+
+		latestPolicySnapshot.Labels[fleetv1.IsLatestSnapshotLabel] = strconv.FormatBool(true)
+		if err := r.Client.Update(ctx, latestPolicySnapshot); err != nil {
+			klog.ErrorS(err, "Failed to set the isLatestSnapshot label to true", "clusterPolicySnapshot", klog.KObj(latestPolicySnapshot))
+			return ctrl.Result{}, err
+		}
+	}
+	// create clusterResourceSnapshot
+	// update the status based on the latestPolicySnapshot status
+	// update the status based on the work
+	return ctrl.Result{}, nil
+}
+
+func ownerReference(crp *fleetv1.ClusterResourcePlacement) metav1.OwnerReference {
+	return metav1.OwnerReference{
+		APIVersion:         crp.GroupVersionKind().GroupVersion().String(),
+		Kind:               crp.GroupVersionKind().Kind,
+		Name:               crp.GetName(),
+		UID:                crp.GetUID(),
+		BlockOwnerDeletion: pointer.Bool(true),
+		Controller:         pointer.Bool(true),
+	}
+}
+
+// lookupLatestClusterPolicySnapshot finds the latest snapshots and its policy index.
+// There will be only one active policy snapshot if exists.
+// It first checks whether there is an active policy snapshot.
+// If not, it finds the one whose policyIndex label is the largest.
+// Return error when 1) cannot list the snapshots 2) there are more than one active policy snapshots 3) snapshot has the
+// invalid label value.
+// 2 & 3 should never happen.
+func (r *Reconciler) lookupLatestClusterPolicySnapshot(ctx context.Context, crp *fleetv1.ClusterResourcePlacement) (*fleetv1.ClusterPolicySnapshot, int, error) {
+	snapshotList := &fleetv1.ClusterPolicySnapShotList{}
+	latestSnapshotLabelMatcher := client.MatchingLabels{
+		fleetv1.CRPTrackingLabel:      crp.Name,
+		fleetv1.IsLatestSnapshotLabel: strconv.FormatBool(true),
+	}
+	crpKObj := klog.KObj(crp)
+	if err := r.Client.List(ctx, snapshotList, latestSnapshotLabelMatcher); err != nil {
+		klog.ErrorS(err, "Failed to list active clusterPolicySnapshots", "clusterResourcePlacement", crpKObj)
+		return nil, -1, err
+	}
+	if len(snapshotList.Items) == 1 {
+		policyIndex, err := parsePolicyIndexFromLabel(&snapshotList.Items[0])
+		if err != nil {
+			return nil, -1, err
+		}
+		return &snapshotList.Items[0], policyIndex, nil
+	} else if len(snapshotList.Items) > 1 {
+		// It means there are multiple active snapshots and should never happen.
+		err := fmt.Errorf("there are %d active clusterPolicySnapshots owned by clusterResourcePlacement %v", len(snapshotList.Items), crp.Name)
+		klog.ErrorS(err, "It should never happen", "clusterResourcePlacement", crpKObj)
+		return nil, -1, err
+	}
+	// When there are no active snapshots, find the one who has the largest policy index.
+	if err := r.Client.List(ctx, snapshotList, client.MatchingLabels{fleetv1.CRPTrackingLabel: crp.Name}); err != nil {
+		klog.ErrorS(err, "Failed to list all clusterPolicySnapshots", "clusterResourcePlacement", crpKObj)
+		return nil, -1, err
+	}
+	if len(snapshotList.Items) == 0 {
+		return nil, -1, nil
```

Comment:
This means our index starts from 0, first policy snapshot will be  "crpxxxx-0". Is this what you intended?

Reply:
yeah, will add a comment to indicate that